zds/tutorialv2/views/statistics.py

```diff
@@ -229,27 +243,30 @@ def get_context_data(self, **kwargs):
         keywords = {}
         report_field = [("nb_uniq_visitors", False), ("nb_hits", False), ("avg_time_on_page", True)]
 
-        for url in urls:
-            all_stats = self.get_all_stats(url, start_date, end_date)
-            cumul_stats = self.get_cumulative(all_stats)
-            all_referrers = self.get_all_refs(url, start_date, end_date, "Referrers.getWebsites")
-            all_type_referrers = self.get_all_refs(url, start_date, end_date, "Referrers.getReferrerType")
-            all_keywords = self.get_all_refs(url, start_date, end_date, "Referrers.getKeywords")
+        # Each function sends only one bulk request for all the urls
+        # Each variable is a list of dictionnaries (one for each url)
+        all_stats = self.get_all_stats(urls, start_date, end_date)
+        all_referrers = self.get_all_refs(urls, start_date, end_date, "Referrers.getWebsites")
+        all_type_referrers = self.get_all_refs(urls, start_date, end_date, "Referrers.getReferrerType")
+        all_keywords = self.get_all_refs(urls, start_date, end_date, "Referrers.getKeywords")
+
+        for index, url in enumerate(urls):
```

Comment:
je l'aurais nommé `url_hierarchy_index` pour rappeler que le principal besoin c'est les tutos big et moyens

Reply:
Qu'est-ce que tu aurais nommé `url_hierarchy_index` ?

Reply:
la variable index

Reply:
Hum, pour être honnête je pense que ça rendrait le code plus confus qu'autre chose. On a `urls` qui est une liste de `NamedUrl` sur laquelle on itère, à chaque itération on note l'index et l'élément en question. Il ne s'agit pas de l'index dans la hiérarchie du contenu (ce qui est comme ça que je comprends `url_hierarchy_index`).